sdk/dotnet/Pulumi/Core/Urn.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright 2016-2018, Pulumi Corporation
+
+#nullable enable
+
+using System;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// An automatically generated logical URN, used to stably identify resources.
+    /// </summary>
+    public sealed class Urn : IEquatable<Urn>

```

Comment:
Should we consider overloading `operator==` and `operator!=`?

Reply:
Done.